src/Type/ArrayType/ArrayType.php

```diff
@@ -422,4 +422,19 @@ public static function uniqueValuesByCallback(array $haystack, Closure $callback
 		return $result;
 	}
 
+	/**
+	 * Mimics array_key_exists
+	 *
+	 * @param array $haystack
+	 * @param integer|string $key
+	 * @return boolean key exists
+	 */
+	public static function hasKey(array $haystack, $key)
+	{
+		if (!array_key_exists($key, $haystack)) {
```

Comment:
It should. It's residue from idea I had about testing whether the key is strictly the same - which does not make sense as array converts the values anyway :( (i.e. will make `1` from `"1"`)